### DISCUSSION TITLE
Get the current mode from the terminal

### DISCUSSION THREAD

**Discussion by tesses** [pit-ray/win-vind#97] [Tips]
I want to make editor's key-bindings when a specific mode is active.

A command in the shell like that probably it can solve this problem:
`path/to/exe/win-vind.exe --status`
Response:
`
{"mode": "en", "version": "4.3"}
`

**Comment by tesses** [comment]
No ok probably this feature is useless without a way to disable absorbing mode in unused key-binding

**Comment by pit-ray** [comment]
We can run the command from a sub-process with `$ win-vind -f easy_click_left`, but I see no way to know the current status.
It might be helpful if the status and version were indicated.

**version**
```sh
$ win-vind --version
4.3.2 portable
```
**status**
```sh
$ win-vind --status
mode: gui-normal
```

Is there any other information needed for the status other than the mode?
If so, it would be useful to be able to extract values as follows
```sh
$ win-vind --status mode
gui-normal
```

Thanks.

**Reply by zzeitt** [reply]
Knowing the current status is useful for me as sometimes when I leave my computer and return to it after a while, I hardly remember what status `win-vin` is in. So I have to press <kbd>Esc</kbd>+<kbd>Left</kbd> to ensure that I am in GUI NORMAL MODE. However, in vim editor (on bash), when user switch to INSERT MODE or VISUAL MODE, there will be a long-standing reminder text showing '-- INSERT --' or '-- VISUAL --', which is useful.

Maybe we could add a similar function to display current status. The implementation would be similar to displaying commands on screen, but setting the fadeout time to 'infinite'.

**Reply by pit-ray** [reply]
Please try `set cmd_fadeout=3600`.

**Comment by tesses** [comment]
Yeah, probably that's a better sintax.

It would be also cool to give the opportunity to execute a command in command mode from the shell

For example:
`$ win-vind --command :arrange`

Thanks

**Reply by pit-ray** [reply]
It is useful to run commands from the terminal. Very good idea!

Now then, I will implement `--version`, `--status`, and `--command`.

Thanks.

**Comment by pit-ray** [comment]
added #101
